src/middlewared/middlewared/etc_files/local/netdata/alarms.py

```diff
@@ -1,26 +1,38 @@
+import os
 import re
 
 
 async def render(service, middleware):
     def update_alarms(alarms):
-        for key, alarm in alarms.items():
-
-            # These are valid alarms and their respective config files exist as well
-            if alarm.get('path'):
-                with open(alarm['path'], 'r+') as f:
+        for path, alarm_dict in alarms.items():
+            if os.path.exists(path):
+                with open(path, 'r') as f:
                     content = f.read()
-                    f.seek(0)
-                    f.write(
-                        re.sub(
-                            fr'(alarm: {key}[\s\S]*?to: )(.*)',
-                            fr'\1{"silent" if not alarm["enabled"] else "sysadmin"}',
-                            content
-                        )
+
+                alarm = None
+                for key, alarm in alarm_dict.items():
+                    content = re.sub(
+                        fr'(alarm: {key}[\s\S]*?to: )(.*)',
+                        fr'\1{"silent" if not alarm["enabled"] else "sysadmin"}',
+                        content
                     )
-                    f.truncate()
+                # Better safe then sorry
+                if alarm:
```

Comment:
Is it guaranteed the content changed?

If re.sub didnt change the content, we shouldnt need to write the file, and instead make sure it doesnt exist?